src/imap-client.js

```diff
@@ -919,9 +921,15 @@
 
         if (options.answered === true) {
             add.push(ANSWERED_FLAG);
-        } else if (options.answered === true) {
+        } else if (options.answered === false) {
             remove.push(ANSWERED_FLAG);
         }
+        
+        if (options.answered === true) {
```

Comment:
I just thought about it, it would make more sense to just run move to push a message to trash than flag it